frontend/src/pages/profile.js

```diff
@@ -1,20 +1,66 @@
-import React, { useContext } from "react";
+import React, { useContext, useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { useParams } from "react-router-dom";
 import Container from "react-bootstrap/Container";
 import Row from "react-bootstrap/Row";
 import Col from "react-bootstrap/Col";
 
+import { setProfile } from "../store/profile";
 import ProfileOverview from "../components/profileOverview";
 import ActivitiesPanel from "../components/activitiesPanel";
+import { UserContext } from "../utils/userContext";
+import * as profileService from "../services/profileService";
 
 const Profile = () => {
+  const { id } = useParams();
+  const { user } = useContext(UserContext);
+  const dispatch = useDispatch();
+  const profileState = useSelector((state) => state.profile.value);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      let follow;
+      const followers = await profileService.getFollowers(id ? id : user.id);
+      const followings = await profileService.getFollowings(id ? id : user.id);
+      const currentProfile = await profileService.getCurrentProfile();
+      const { data } = id
+        ? await profileService.getOtherProfile(id)
+        : currentProfile;
```

Comment:
This is not a good practice. 4 different requests on a single page. 
The problem is that if the user has slow internet connection, it will take forever to load.
Find a way to optimize your request. It would be better if 1 request per page.

What you can do:
- Create api to merge all request in 1 api request
- followers and followings api can be transferred to activities panel to each respective body. Make it dynamic to fetch followers or following depends which is opened
 So you have 2 Pros on this approach
 1. Only fetch data when it is needed
 2. Fetch the most updated data, maybe there is new followers in a second for example